pretyping/pretyping.ml

```diff
@@ -902,9 +902,13 @@ let rec pretype k0 resolve_tc (tycon : type_constraint) (env : ExtraEnv.t) evdre
       user_err ?loc  (str "Destructing let is only for inductive types" ++
 	str " with one constructor.");
     let cs = cstrs.(0) in
+    let valname = Names.Name.print na in
+    let indtype = (fst (dest_ind_family indf)) in
+    let typename = Printer.pr_pinductive env.ExtraEnv.env indtype  in
```

Comment:
Don't use Printer in pretyping.
Maybe the error should be added to Pretype_errors.

Reply:
spot on

Reply:
If I can't use `Printer`, then I need to extend `Pretype_errors` and plumb this to print the info?

Reply:
That should do the trick.